tests/test_indexing.py

```diff
@@ -19,7 +19,10 @@ def test_index():
 def test_full_window():
     with rasterio.open('tests/data/RGB.byte.tif') as src:
         left, bottom, right, top = src.bounds
-        assert src.window(left, bottom, right, top) == tuple(zip((0, 0), src.shape))
+        assert src.window(
```

Comment:
Yeah, I remembered looking at that one, trying to figure out how to make it more readable. Your solution is much better.